### PR TITLE
Fix API changes to preserve backward compatibility

### PR DESCRIPTION
**Pull request by tharindu-bandara**
### Proposed changes in this pull request

- Fixes [product-is/issues/5921](https://github.com/wso2/product-is/issues/5921).
- Introduce the previous method signature, and refactor the changed code as an overloaded method.

### When should this PR be merged

- ASAP

#### Documentation
- N/A

### CODE REVIEW

File: components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/openidconnect/OIDCClaimUtil.java [hunk 1]
```diff
@@ -98,6 +98,35 @@ private static String getLocalRoleName(RoleMapping roleMapping) {
         return roleMapping.getLocalRole().getLocalRoleName();
     }
 
+    /**
+     * Filter user claims based on consent with the highest priority {@link OpenIDConnectClaimFilter}. Consent based
+     * user claims filtering can be configured at the global level.
+     */
+    public static Map<String, Object> filterUserClaimsBasedOnConsent(Map<String, Object> userClaims,
```
Comment (malithie) [inline components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/openidconnect/OIDCClaimUtil.java]: If this method is not recommended to use and is in place just to ensure backward compatibility better if we can deprecate it

Comment (mefarazath) [reply components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/openidconnect/OIDCClaimUtil.java]: Better to indicate the method to use instead as well.

File: components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/openidconnect/OIDCClaimUtil.java [hunk 2]
```diff
@@ -98,6 +98,36 @@ private static String getLocalRoleName(RoleMapping roleMapping) {
...
+                                                                     String grantType) {
+
+        if (isConsentBasedClaimFilteringApplicable(grantType)) {
+            return OpenIDConnectServiceComponentHolder.getInstance()
+                    .getHighestPriorityOpenIDConnectClaimFilter()
+                    .getClaimsFilteredByUserConsent(userClaims, authenticatedUser, clientId, spTenantDomain);
+        } else {
+            if (log.isDebugEnabled()) {
+                String msg = "Filtering user claims based on consent skipped for grant type:%s. Returning original " +
+                        "user claims for user:%s, for clientId:%s of tenantDomain:%s";
```
Comment (GDRDABARERA) [inline components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/openidconnect/OIDCClaimUtil.java]: shall we add a space after :

File: components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/openidconnect/OIDCClaimUtil.java [hunk 3]
```diff
@@ -98,6 +98,36 @@ private static String getLocalRoleName(RoleMapping roleMapping) {
         return roleMapping.getLocalRole().getLocalRoleName();
     }
 
+    @Deprecated
```
Comment (GDRDABARERA) [inline components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/openidconnect/OIDCClaimUtil.java]: Please mentioned the introduced method if this is deprecated

Comment (tharindu-bandara) [reply components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/openidconnect/OIDCClaimUtil.java]: +1. Updated the Javadoc comment.

File: components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/openidconnect/OIDCClaimUtil.java (hunk 2 above)
Comment (tharindu-bandara) [reply components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/openidconnect/OIDCClaimUtil.java]: Fixed now.

**Pull request merged**